packages/create-pantheon-decoupled-kit/src/index.ts

```diff
@@ -85,11 +85,29 @@ export const main = async (
 	// If no generators are found in positional params
 	// ask which generators should be run
 	if (!foundGenerators || !foundGenerators.length) {
-		const whichGenerators: QuestionCollection<{ generators: string[] }> = {
+		// check if a cmsType was provided
+		const cmsType: string | null = args.cmsType
+			? new String(args.cmsType).toLowerCase()
+			: null;
+
+		let suggestedGeneratorNames: string[] | null;
+
+		// filter generators if a cmsType was provided
+		if (cmsType) {
+			const suggestedGenerators = generators.filter((generator) => {
+				return generator.cmsType === cmsType || generator.cmsType === 'any';
+			});
+			suggestedGeneratorNames = suggestedGenerators.map(({ name }) => name);
+		}
```

Comment:
This can be a bit more concise, and the name of the generator names could be more specific.
```suggestion
		const generatorsOfCmsType = generators
			.filter((generator) => {
				return isString(args.cmsType)
					? generator.cmsType === args.cmsType.toLowerCase() ||
							generator.cmsType === 'any'
					: true;
			})
			.map(({ name }) => name);
```

Reply:
Used your suggestion as a base and added a few cases to handle error messaging and avoid returning only generators with type `any` if an incorrect `cmsType` was provided